packages/@atjson/renderer-hir/test/renderer-hir-test.ts

```diff
@@ -235,12 +246,170 @@ describe("@atjson/renderer-hir", () => {
         return `_${words.join("")}_`;
       }
 
-      *root(): IterableIterator<any> {
+      *root(): Iterator<void, string, string[]> {
         let rawText: string[] = yield;
         return rawText.join("");
       }
     }
 
     expect(SlackRenderer.render(doc)).toBe("> *I am _very_ excited*");
   });
+
+  describe("slices", () => {
+    test("top level keys are reified", () => {
+      let doc = new TestSource({
+        content:
+          "The radio is stuck right now. Everything sound the same. As far as video-wise everything look the same. So we coming in to change the whole thing. Missy Elliott",
+        annotations: [
+          new BlockQuote({
+            id: "a1",
+            start: 0,
+            end: 146,
+            attributes: { credit: "a2" },
```

Comment:
I would like it if we could differentiate slice references from arbitrary strings in some way. maybe a url or some similar scheme?
 - `slice:a2`
 - `slice://a2`
 - `#a2`
idk these are just my first blush ideas

Reply:
yeah, this is something that's been floating in my head, specifically whether we should have a mongoish `ObjectId` thing that is used throughout atjson to refer to ids

Reply:
Annotations automatically get ids if they're not already present, so that's definitely an option. I like the #a2 approach, as it is compatible with generic URIs; for example, something like https://example.com/mydoc.atjson#a2 would also work (not suggesting we broach that particular complexity hurdle, just observing that this wouldn't preclude that option).

That said, I'm not sure that it's sufficiently distinct to satisfy @colin-alexa's very salient concern. Two thoughts/questions:

1. Are slices special in some way? Can any annotation be a slice?
2. Is there some other AST-ish way that we can indicate the intent is a slice?

e.g., ```credit: slice("a2")``` would feel more robust than just implicitly keying off even a namespaced string, but obviously that doesn't work. In Automerge, there's a way (or will be 😂) to build new types that would address this, but that doesn't really apply here.. 🙃

Reply:
What about just `attributes: { credit: { slice: "a2" } }`?

Reply:
Oh, so the concern here I think is a security concern. There's a possibility that there's magic strings that can suddenly cause text to be included in rendering.

Slices are special in _only_ the rendering step, where they are taken out and put into fragment HIR nodes for rendering.

There's a weird edge case with React rendering that I've encountered with implementing this shallowly (at the edge) where there's some terrible performance. This could be seen very clearly with this example https://wikiatjson.glitch.me/wiki/Boston. This _should_ be quick to render, but has so many slices that something goes _very wrong_ with React, trying to render nested trees.

Reply:
so, thinking this through... I think there needs to be a concrete serialization into JSON that doesn't cause security issues (or at least mitigates them). @blaine the solution you provided sounds good. within atjson runtime, I think to be able to distinguish this type, there may need to be some runtime object to refer to references.

Reply:
@tim-evans I think you had come to a conclusion that the current approach is satisfactory for now? Can you update this comment with that?